src/psyclone/tests/domain/lfric/kern_call_invoke_arg_list_test.py

```diff
@@ -0,0 +1,97 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2022, Science and Technology Facilities Council
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: A. R. Porter, STFC Daresbury Lab
+
+''' Module containing pytest tests for the xxx class. '''
+
+import pytest
+
+from psyclone.domain.lfric.kern_call_invoke_arg_list import (
+    KernCallInvokeArgList)
+from psyclone.psyir.symbols import SymbolTable, DataTypeSymbol, DeferredType
+
+
+def test_kcial_construct(dynkern):
+    ''' Tests for the KernCallInvokeArgList constructor. '''
+    with pytest.raises(TypeError) as err:
+        KernCallInvokeArgList(dynkern, None)
+    assert ("Argument 'symbol_table' must be a SymbolTable instance but got "
+            "'NoneType'" in str(err.value))
+    obj = KernCallInvokeArgList(dynkern, SymbolTable())
+    assert obj.fields == []
```

Comment:
Triggers 'use-implicit-booleaness-not-comparison`, but I think that should just be disabled (since we otherwise would also accept None, or 0, ... as correct). If you agree and are happy to do this, maybe add this to our psyclone/tests/.pylintrc file - otherwise ignore my comment.

Reply:
Good idea - done.